### PR TITLE
Add SAML exception

### PR DESCRIPTION
**Pull request by Borda**
recently crossed this exception and think would be nice to catch it...

cc: @EnricoMi

### CODE REVIEW

File: github/Requester.py [hunk 1]
```diff
@@ -542,20 +541,20 @@ def createException(
...
-        elif status == 403 and message.startswith(
-            "missing or invalid user agent string"
-        ):
-            exc = GithubException.BadUserAgentException
-        elif status == 403 and cls.isRateLimitError(message):
-            exc = GithubException.RateLimitExceededException
+        if status == 401:
+            if message == "bad credentials":
+                exc = GithubException.BadCredentialsException
+            if "SAML enforcement" in message:
```
Comment (trim21) [inline github/Requester.py:547]: message is in lower case

**Review by trim21 — CHANGES_REQUESTED** [review]

File: github/Requester.py (hunk 1 above)
Comment (Borda) [reply github/Requester.py:547]:
```suggestion
            if "saml enforcement" in message.lower():
```

File: github/Requester.py [hunk 2]
```diff
@@ -544,7 +544,7 @@ def createException(
         if status == 401:
             if message == "bad credentials":
                 exc = GithubException.BadCredentialsException
-            if "SAML enforcement" in message:
+            if "saml enforcement" in message.lower():
```
Comment (trim21) [inline github/Requester.py:547]: message is already lower

Comment (Borda) [reply github/Requester.py:547]:
```suggestion
            if "saml enforcement" in message:
```

**Review by trim21 — APPROVED** [review]
lgtm

**Comment by EnricoMi** [comment]
Please request review so this flags up for me again.

**Pull request closed without merging**